src/mtcp/mtcp_restart.c

```diff
@@ -788,9 +788,10 @@ unmap_memory_areas_and_restore_vdso(RestoreInfo *rinfo)
     // Since vdso will use randomized addresses (unlike the standard practice
     // for vsyscall), this implies that kernel calls on __x86__ can go through
     // randomized addresses, and so they need special treatment.
-    void *vdso = mtcp_sys_mmap(vdsoStart, vdsoEnd - vdsoStart,
-                               PROT_EXEC | PROT_WRITE | PROT_READ,
-                               MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
+    void *vdso = mmap_fixed_noreplace(vdsoStart, vdsoEnd - vdsoStart,
```

Comment:
REPEATED:
The `MAP_FIXED` is redundant, since `map_fixed_noreplace()` will simply remove the `MAP_FIXED`.
We can keep it here, or remove `MAP_FIXED` with the idea that the name `map_fied_noreplace()` already says it all.
I'm open to either option (keeping it as is, or removing it).  Just a thought.